mlflow/server/js/src/components/RequestStateWrapper.js

```diff
@@ -34,9 +34,9 @@ export class RequestStateWrapper extends Component {
   }
 
   static getDerivedStateFromProps(nextProps) {
-    const shouldRender = nextProps.requests.length ? nextProps.requests.every((r) => {
+    const shouldRender = nextProps.requests.every((r) => {
```

Comment:
Adding the check of `nextProps.requests.length` is to support initial loading from componentDidMount. This will likely break model registry pages which make initial loads in component did mount. We should probably try using `shouldOptimisticallyRender` on the special case?

Reply:
@Zangr thanks for the comment! I'm not entirely sure I follow as the model registry components all seem to pass non-empty arrays of request IDs to `RequestStateWrapper`,  (i.e. the empty length case never applies to them). But using `shouldOptimisticallyRender` here does work, so I'll switch to that instead :)

Reply:
As React triggers `componentDidMount` in "post order traversal", the initial render of RequestStateWrapper will not find any requests. However`[].every(...)` returns true when the base array is empty. That will make the component show before data is ready, which can cause a flash of semi-ready content or NPE.